### DISCUSSION TITLE
Question about using const idx_t & in source code

### DISCUSSION THREAD

**Discussion by xuke-hat** [duckdb/duckdb#7834] [General]
I have been reading DuckDB's source code recently and noticed that some parts of the code use `const idx_t &` as argument instead of `idx_t`. Is there any reason about it?

In my opinion `const idx_t &` will pass the address of the value, requiring the function to load the value from its address. If we just pass `idx_t` by value we can omit the load.

**Comment by Tishj** [comment]
they're both 8 bytes, I don't think there's a real difference in performance
